helm-charts/konk-service/templates/_helpers.tpl

```diff
@@ -81,3 +81,16 @@ dummy
 {{- fail "cert-manager CRD does not appear to be installed" }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Templates konk name
+*/}}
+{{- define "konk-service.konkname" -}}
+{{- if hasSuffix "-konk" .Values.konk.name }}
+{{- .Values.konk.name }}
+{{- else }}
+{{- required "konk name must end in -konk" .Values.notexist }}
```

Comment:
This seems like the wrong place to enforce this. It won't fail until the operator tries to install the KonkService, and by then the Konk already exists. The better place to enforce this is the validation section of the CRD, since that will give immediate visible feedback to the user during installation.

Also, you can use `fail` for this:
```suggestion
{{- fail "konk name must end in -konk" }}
```

Reply:
good idea for the next PR. I wanted to enforce the idea that we're not going to attempt to fix the konk.name inside template code. I want it to fail.

Reply:
The problem is that the template failure gets buried in the CRD status message. It's not an obviously failure. Not very user friendly.